### PR TITLE
feat: add decorator to manually parse cookies

### PR DESCRIPTION
**Pull request by DRoet**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

Hi there,

I added this decorator to help me decrypt cookies/sessions when using `fastify-websocket` (see https://github.com/fastify/fastify-websocket/issues/70) , since wsHandlers don't have access to the `FastifyRequest ` you have to manually decrypt cookies/sessions. I've already submitted a PR to `fastify-session` to be able to decrypt sessions manually here https://github.com/SerayaEryn/fastify-session/pull/128.

However I would like to parse the cookie manually without pulling in the `cookie` package and duplicating `options` logic in two locations, hence this added decorator to make this step a little easier

### CODE REVIEW

File: plugin.js [hunk 1]
```diff
@@ -36,12 +36,16 @@ function fastifyCookieClearCookie (reply, name, options) {
...
+  return cookie.parse(cookieHeader, options)
+}
+
 function onReqHandlerWrapper (options) {
   return function fastifyCookieOnReqHandler (fastifyReq, fastifyRes, done) {
     fastifyReq.cookies = {} // New container per request. Issue #53
     const cookieHeader = fastifyReq.raw.headers.cookie
     if (cookieHeader) {
-      fastifyReq.cookies = cookie.parse(cookieHeader, options)
+      fastifyReq.cookies = fastifyParseCookie(cookieHeader, options)
```
Comment (mcollina) [inline plugin.js:48]: please use the decorator function instead

File: plugin.js [hunk 2]
```diff
@@ -50,6 +54,9 @@ function onReqHandlerWrapper (options) {
 function plugin (fastify, options, next) {
   const secret = options ? options.secret || '' : ''
 
+  fastify.decorate('parseCookie', function parseCookie (cookieHeader) {
+    return fastifyParseCookie(cookieHeader, options)
```
Comment (mcollina) [inline plugin.js:58]: just call cookie.parse here

**Review by jsumners — CHANGES_REQUESTED** [review]
Documentation is missing.

File: plugin.js (hunk 1 above)
Comment (DRoet) [reply plugin.js:48]: Do I have to pass the fastify instance to the onReqHandlerWrapper or is there a better way of doing this?

**Review by mcollina — APPROVED** [review]
lgtm

File: README.md [hunk 3]
```diff
@@ -74,6 +74,12 @@ on the cookie path `/`.
 + `options`: an options object as described in the [cookie serialize][cs]
 documentation. Its optional to pass `options` object
 
+### Manual cookie parse
```
Comment (jsumners) [inline README.md:77]:
```suggestion
### Manual cookie parsing
```

**Review by jsumners — APPROVED** [review]
LGTM with grammar fix

**Review by fox1t — APPROVED** [review]
LGTM

**Comment by DRoet** [comment]
Thank you everyone for the quick review + release!

**Pull request merged**